### PR TITLE
Add vSphere VIP support

### PR DESCRIPTION
**Pull request by mtnbikenc**
Adds the optional install-config variables for APIVIP, IngressVIP, and DNSVIP to enable IPI deployment.

https://issues.redhat.com/browse/CORS-1317

Needs:
- [x] https://github.com/openshift/api/pull/541

### CODE REVIEW

File: pkg/asset/manifests/infrastructure.go [hunk 1]
```diff
@@ -144,6 +143,12 @@ func (i *Infrastructure) Generate(dependencies asset.Parents) error {
 		}
 	case vsphere.Name:
 		config.Status.PlatformStatus.Type = configv1.VSpherePlatformType
+		// Needs https://github.com/openshift/api/pull/541
+		// config.Status.PlatformStatus.VSphere = &configv1.VSpherePlatformStatus{
```
Comment (jcpowermac) [inline pkg/asset/manifests/infrastructure.go]: @mtnbikenc 
Can you add a check here if the strings are empty.  If so we will want the PlatformStatus.VSphere to be nil.
MCO then won't render the bootstrap manifests that is required only for IPI.

Comment (jcpowermac) [reply pkg/asset/manifests/infrastructure.go]: See https://github.com/openshift/machine-config-operator/pull/1373/files#diff-d6f38878757202a810cfa8038e7ae122R272
for additional details.

Comment (mtnbikenc) [reply pkg/asset/manifests/infrastructure.go]: Updated.

File: pkg/asset/ignition/machine/node.go [hunk 2]
```diff
@@ -11,7 +11,8 @@ import (
 	baremetaltypes "github.com/openshift/installer/pkg/types/baremetal"
 	openstacktypes "github.com/openshift/installer/pkg/types/openstack"
 	openstackdefaults "github.com/openshift/installer/pkg/types/openstack/defaults"
-	"github.com/openshift/installer/pkg/types/ovirt"
+	ovirttypes "github.com/openshift/installer/pkg/types/ovirt"
```
Comment (sdodson) [inline pkg/asset/ignition/machine/node.go]: Intentionally included here?

File: pkg/asset/tls/mcscertkey.go [hunk 3]
```diff
@@ -3,14 +3,15 @@ package tls
...
 	"crypto/x509/pkix"
-	"github.com/openshift/installer/pkg/types/ovirt"
 	"net"
 
 	"github.com/openshift/installer/pkg/asset"
 	"github.com/openshift/installer/pkg/asset/installconfig"
 	baremetaltypes "github.com/openshift/installer/pkg/types/baremetal"
 	openstacktypes "github.com/openshift/installer/pkg/types/openstack"
 	openstackdefaults "github.com/openshift/installer/pkg/types/openstack/defaults"
+	ovirttypes "github.com/openshift/installer/pkg/types/ovirt"
```
Comment (sdodson) [inline pkg/asset/tls/mcscertkey.go]: Intentionally included here?

File: pkg/asset/ignition/machine/node.go (hunk 2 above)
Comment (sdodson) [reply pkg/asset/ignition/machine/node.go]: @rgolangh FYI if so.

Comment (mtnbikenc) [reply pkg/asset/ignition/machine/node.go]: I was standardizing the imports to use `*types`.  I can back that out if it is an issue.

File: pkg/asset/tls/mcscertkey.go (hunk 3 above)
Comment (mtnbikenc) [reply pkg/asset/tls/mcscertkey.go]: I was standardizing the imports to use `*types`.  I can back that out if it is an issue.

File: pkg/asset/ignition/machine/node.go (hunk 2 above)
Comment (sdodson) [reply pkg/asset/ignition/machine/node.go]: lets leave it, just wanted to make sure that it wasn't an oversight

**Comment by jcpowermac** [comment]
```
level=error msg="Error: ResourceLimitExceeded: Task limit exceeded"
```

/test e2e-aws-upgrade

**Comment by jcpowermac** [comment]
Looks like a lot of AWS issues
```
level=error msg="Error: Error waiting for AMI (ami-072b03e6046c527e0) to be ready: timeout while waiting for state to become 'available' (last state: 'pending', timeout: 40m0s)"
```
/test e2e-aws-upgrade

**Comment by jcpowermac** [comment]
Keep getting the same AMI error:
```
level=error msg="Error: Error waiting for AMI (ami-0ba76cd1f4f3993da) to be ready: timeout while waiting for state to become 'available' (last state: 'pending', timeout: 40m0s)"
```

**Comment by vrutkovs** [comment]
This breaks vSphere UPI: generated URL for workers / masters is `https://:22623/config/worker.ign` :/

File: pkg/asset/ignition/machine/node.go [hunk 4]
```diff
@@ -30,8 +31,10 @@ func pointerIgnitionConfig(installConfig *types.InstallConfig, rootCA []byte, ro
 		} else {
 			ignitionHost = fmt.Sprintf("api-int.%s:22623", installConfig.ClusterDomain())
 		}
-	case ovirt.Name:
+	case ovirttypes.Name:
 		ignitionHost = fmt.Sprintf("%s:22623", installConfig.Ovirt.APIVIP)
+	case vspheretypes.Name:
```
Comment (jcpowermac) [inline pkg/asset/ignition/machine/node.go]: @mtnbikenc @vrutkovs right here is where ignition is broken.  Need to add a check for empty string APIVIP

Comment (mtnbikenc) [reply pkg/asset/ignition/machine/node.go]: https://github.com/openshift/installer/pull/2956

**Pull request merged**